listenbrainz/webserver/static/js/src/PinRecordingModal.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from "react";
+import { get as _get } from "lodash";
+import GlobalAppContext from "./GlobalAppContext";
+
+export type PinRecordingModalProps = {
+  recordingToPin?: Listen;
+  isCurrentUser: Boolean;
+  newAlert: (
+    alertType: AlertType,
+    title: string,
+    message: string | JSX.Element
+  ) => void;
+};
+
+export interface PinRecordingModalState {
+  blurbContent?: string;
+}
+
+export default class PinRecordingModal extends React.Component<
+  PinRecordingModalProps,
+  PinRecordingModalState
+> {
+  static contextType = GlobalAppContext;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+  readonly maxBlurbContentLength = 280;
+
+  constructor(props: PinRecordingModalProps) {
+    super(props);
+    this.state = { blurbContent: undefined };
+  }
+
+  submitPinRecording = async () => {
+    const { recordingToPin, isCurrentUser, newAlert } = this.props;
+    const { blurbContent } = this.state;
+    const { APIService, currentUser } = this.context;
+
+    if (isCurrentUser && currentUser?.auth_token) {
+      const recordingMSID = _get(
+        recordingToPin,
+        "track_metadata.additional_info.recording_msid"
+      );
+      const recordingMBID = _get(
+        recordingToPin,
+        "track_metadata.additional_info.recording_mbid"
+      );
+
+      try {
+        const status = await APIService.submitPinRecording(
+          currentUser.auth_token,
+          recordingMSID,
+          recordingMBID || undefined,
+          blurbContent || undefined
+        );
+        if (status === 200) {
+          newAlert(
+            "success",
+            `You pinned a recording!`,
+            `${recordingToPin?.track_metadata.artist_name} - ${recordingToPin?.track_metadata.track_name}`
+          );
+          this.setState({ blurbContent: undefined });
+        }
+      } catch (error) {
+        this.handleError(error, "Error while pinning recording");
+      }
+    }
+  };
+
+  handleBlurbInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    event.preventDefault();
+    const input = event.target.value.replace(/\s\s+/g, " "); // remove line breaks and excessive spaces
```

Comment:
Is there a reason we don't allow line breaks?

Reply:
It just makes the cards look uniform when they eventually get rendered back to the user, otherwise we could get a blurb content with nothing but line breaks. Restricting it in the input rather is better imo because the user can see how it will look when it gets rendered.

Twitter allows one line break at a time in tweets, maybe we could do that if this is too restrictive?

Reply:
No I think it's okay for now, let's iterate on it later if we need to.